standards/git-workflow.md

```diff
@@ -0,0 +1,199 @@
+# Git workflow
+
+All repos MUST identify their "Git workflow" (aka "Merge strategy", "Steps for promoting a feature") in their README. A Git workflow identifies:
+
+ * the persistent branches (e.g. "development", "qa", "master"/"main"\*), including what environment they are deployed to if any
+ * what branch to cut feature branches from
+ * whether and how one bumps the component version (if it deviates from [the standard](./versioning.md#typical-workflow))
+ * any deviations from [our typical CI/CD configuration](https://github.com/NYPL/engineering-general/blob/updating-header/standards/travis-ci.md)
+
+Repos SHOULD link to the relevant Git workflow in this document and need not say anything more about the procedure if this document adequately describes it.
+
+For example:
+
+```
+## Contributing
+
+This repo uses the [Development-QA-Main Git Workflow](https://github.com/NYPL/engineering-general/tree/master/standards/git-workflow#development-qa-main)
+```
+
+## A note on master/main
+
+Repos should use "main" over "master". The workflows described in this document favor the term "main" over "master" anticipiting that all NYPL repos will eventually make that shift.
+
+## Common worklows
+
+The following are common Git workflows, named here so they can be easily referenced by the components that use them.
+
+### Development-QA-Main
+<a id="development-qa-master"></a>
+
+![Git Workflow: Development-QA-Main diagram](./assets/git-workflow-development-qa-main.png)
+
+| Branch        | Environment | Cut features from this branch | Version tag this branch |
+|---------------|-------------|-------------------------------|-------------------------|
+| `development` | Development | ✅                            |                         |
+| `qa`          | QA          |                               |                         |
+| `main`        | Production  |                               | ✅                      |
+
+1. Create feature branch off `development`
+1. Compute next logical version and update `README.md`, `CHANGELOG.md`, `package.json`, etc.
+1. Create PR against `development`
+1. After review, merge `development` > `qa`
+1. After QC signoff, merge `qa` > `main`
+1. Git tag `main` with new version number.
+
+Example components:
+ * https://github.com/NYPL/dgx-blogs
+ * https://github.com/NYPL/dgx-homepage
+ * https://github.com/NYPL/nypl-dgx-react-header
+ * https://github.com/NYPL-discovery/discovery-front-end
+ * https://github.com/NYPL/dgx-booklists
+ * https://github.com/NYPL/staff-picks
+ * https://github.com/NYPL/dgx-new-arrivals
+ * https://github.com/NYPL-discovery/sierra-retriever
+ * https://github.com/NYPL/MyLibraryNYCApp (No version tagging)
+ * https://github.com/NYPL/locations-app (Has a `production` branch solely for storing production app config.)
```

Comment:
I think we can remove deprecated repos. If so, then `locations-app`, `staff-picks`, and `dgx-blogs`.